src/Symfony/Component/PasswordHasher/Hasher/PasswordHasherFactory.php

```diff
@@ -38,7 +40,7 @@ public function __construct(array $passwordHashers)
     /**
      * {@inheritdoc}
      */
-    public function getPasswordHasher($user): PasswordHasherInterface
+    public function getPasswordHasher(UserInterface|string|PasswordAuthenticatedUserInterface $user): PasswordHasherInterface
```

Comment:
This misses `PasswordHasherAwareInterface` (unless we expect to support that only in combination with `PasswordAuthenticatedUserInterface`)

Reply:
`UserInterface` can be removed also.
```suggestion
    public function getPasswordHasher(string|PasswordAuthenticatedUserInterface|PasswordHasherAwareInterface $user): PasswordHasherInterface
```